agent/consul/server.go

```diff
@@ -1010,20 +1010,56 @@ func (s *Server) WANMembers() []serf.Member {
 }
 
 // RemoveFailedNode is used to remove a failed node from the cluster
-func (s *Server) RemoveFailedNode(node string) error {
-	if err := s.serfLAN.RemoveFailedNode(node); err != nil {
+// We must remove a node from both serf LAN & WAN if present and also
+// check for the prune flag to see if it should be completely deleted
+func (s *Server) RemoveFailedNode(node string, prune bool) error {
+	if prune {
```

Comment:
I am very confused by the control flow in this function now.

Reply:
I cleaned this up a bit, let me know if you still are confused and I can give some insight on my thought process. Perhaps I'm approaching this wrong? 🤷‍♀